realtime/utils.js

```diff
@@ -400,7 +401,15 @@ function isIpWhitelisted(addr, whitelist) {
  */
 function attachAspectSubject(sample, useSampleStore, subjectModel,
   aspectModel) {
-  const nameParts = sample.name.split('|');
+  let nameParts;
```

Comment:
Can we do something like 
if (!sample.name) {
 log error and return undefined 
}

and in "redisPublisher.js" check for "falsey" in line 134 before calling publishObject ?

Reply:
I mean we can do but this is just for debugging that's why. If this is the final solution then we can use this.

Reply:
Let me know what do you think? @shriramshankar

Reply:
I think Ian's comments kind of answers your question.